components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java

```diff
@@ -118,19 +118,33 @@ private void persist(Object dataObj, String resourcePath) throws AutoScalerExcep
     }
 
     public void persistAutoscalerPolicy(AutoscalePolicy autoscalePolicy) {
-        String resourcePath = AutoscalerConstants.AUTOSCALER_RESOURCE + AutoscalerConstants.AS_POLICY_RESOURCE + "/" + autoscalePolicy.getId();
-        persist(autoscalePolicy, resourcePath);
-        if (log.isDebugEnabled()) {
-            log.debug(String.format("Autoscaler policy written to registry: [id] %s [name] %s [description] %s",
-                    autoscalePolicy.getId(), autoscalePolicy.getDisplayName(), autoscalePolicy.getDescription()));
+        try {
+            String resourcePath = AutoscalerConstants.AUTOSCALER_RESOURCE +
+                    AutoscalerConstants.AS_POLICY_RESOURCE + "/" + autoscalePolicy.getId();
+
+            persist(autoscalePolicy, resourcePath);
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Autoscaler policy written to registry: [id] %s [name] %s [description] %s",
+                        autoscalePolicy.getId(), autoscalePolicy.getDisplayName(), autoscalePolicy.getDescription()));
+            }
+        } catch (Exception e) {
+            log.error((String.format("Unable to persist autoscaler policy [autoscaler-policy-id] %s",
+                    autoscalePolicy.getId())), e);
```

Comment:
@Vishanth This pull request was open for quite some time now. Shall we proceed further on this? If you could try to re-create the original issue and verify whether this change fix that problem it would be great!
Thanks